concordium-cis2/src/lib.rs

```diff
@@ -1120,6 +1132,163 @@ pub struct OnReceivingCis2Params<T: IsTokenId, A: IsTokenAmount> {
     pub data:     AdditionalData,
 }
 
+/// Identifier for a smart contract standard.
+/// Consists of a string of ASCII characters up to a length of 255.
+///
+/// See [StandardIdentifierOwned] for the owned version.
+#[derive(Debug, Serial, PartialEq, Eq)]
+pub struct StandardIdentifier<'a> {
+    #[concordium(size_length = 1)]
+    id: &'a str,
+}
+
+/// String is not a valid standard identifier. Ensure the length is less than
+/// 256 and it only contains ASCII characters.
+#[derive(Default)]
+pub struct InvalidStandardIdentifierError;
+
+impl<'a> StandardIdentifier<'a> {
+    /// Validate and construct a standard identifier.
+    pub fn new(id: &'a str) -> Result<Self, InvalidStandardIdentifierError> {
+        if id.len() > 255 || !id.is_ascii() {
+            Err(InvalidStandardIdentifierError)
+        } else {
+            Ok(Self {
+                id,
+            })
+        }
+    }
+
+    /// Construct a standard identifier without validation.
+    pub const fn new_unchecked(id: &'a str) -> Self {
+        Self {
+            id,
+        }
+    }
+
+    /// Convert to owned standard identifier.
+    pub fn to_owned(&self) -> StandardIdentifierOwned {
+        StandardIdentifierOwned::new_unchecked(self.id.to_string())
+    }
```

Comment:
Would it make sense to instead implement https://doc.rust-lang.org/std/borrow/trait.ToOwned.html ?